capstone/src/components/User/Adventurer/Preference/FilterMenu/FilterMenu.jsx

```diff
@@ -0,0 +1,37 @@
+import { ChevronDownIcon } from '@chakra-ui/icons';
+import {
+  Menu, MenuButton, Button, MenuList, MenuItem,
+} from '@chakra-ui/react';
+
+// inactivePreferences={inactivePreferences}
+//           setInactivePreferences={setInactivePreferences}
+//           setActivePreferencesIDs={setActivePreferencesIDs}
+function handleAddition(objectId, inactivePreferences, setInactivePreferences, activePreferencesIDs, setActivePreferencesIDs) {
+  setActivePreferencesIDs([...activePreferencesIDs, objectId]);
+  setInactivePreferences(
+    inactivePreferences.filter((preferences) => preferences.objectId !== objectId),
+  );
+}
+
+export default function FilterMenu({
+  inactivePreferences, setInactivePreferences, activePreferencesIDs, setActivePreferencesIDs,
+}) {
```

Comment:
This function would make more sense within `FilterMenu` so you don't have to pass through the arguments - they would already available in that scope.

```
export default function FilterMenu({...}) {
  function handleAddition(objectID) {...}
  ...
}
```

Reply:
Great! That makes so much sense. I am still trying to figure out about dumb and smart components, since I tried to not give FilterMenu that much power by taking the function out of the component.
I will read more about dumb vs smart.

Reply:
Maybe we can put aside the "dumb vs smart" components for now :) it's mainly about sensible data flow, but problems related to that will come up organically.